src/js/module.js

```diff
@@ -262,6 +269,35 @@ iotjs_module_t.load = function(id, parent) {
 };
 
 
+function _makeRequireFunction(mod) {
+  var Module = mod.constructor;
+  function require(id) {
+    return mod.require(id);
+  }
+
+  function _resolve(request) {
+    if (!request || typeof request !== 'string') {
```

Comment:
Shall validation checks of request be done in `resolveModPath`?

Reply:
It is not very rigorous: 
```js
  function _resolve(request) {
    if (!request || typeof request !== 'string') {
      throw new TypeError('module must be a non-null string');
    }

    if (process.builtin_modules[request]) {
      return request;
    }

    var path = Module.resolveModPath(request, mod);
    if (!path) {
      throw new Error('Module not found: ' + request);
    }
    return path;
  }
```
before I invoke `resolveModPath`, I will do this `process.builtin_modules[request]`, so this arg must be non-null string.